### PR TITLE
Add registration settings

### PR DESCRIPTION
**Pull request by tomerlevi444**

### CODE REVIEW

File: KlarnaOnDemand/KODRegistrationViewController.h [hunk 1]
```diff
@@ -46,6 +47,8 @@
  */
 @interface KODRegistrationViewController : KODWebViewController
 
+@property (strong, nonatomic) KODRegistrationSettings *registrationSettings;
```
Comment (lilach) [inline KlarnaOnDemand/KODRegistrationViewController.h]: move to the .m file in extension to make private

File: KlarnaOnDemand/KODRegistrationViewController.h [hunk 2]
```diff
@@ -55,4 +58,6 @@
  */
 - (id)initWithDelegate:(id<KODRegistrationViewControllerDelegate>)delegate;
 
+- (id)initWithDelegate:(id<KODRegistrationViewControllerDelegate>)delegate andRegistrationSettings: (KODRegistrationSettings *) registrationSettings;
```
Comment (lilach) [inline KlarnaOnDemand/KODRegistrationViewController.h]: add documentation

File: KlarnaOnDemand/KODUrl.m [hunk 3]
```diff
@@ -15,6 +16,16 @@ + (NSURL *)registrationUrl {
...
                    [self urlEncodeWithParam:publicKeyBase64Str]];
+
+  if(registrationSettings != nil) {
+    if(registrationSettings.confirmedUserDataId != nil) {
+      url = [NSString stringWithFormat:@"%@&confirmed_user_data_id=%@", url, registrationSettings.confirmedUserDataId];
+    }
+    if(registrationSettings.prefillPhoneNumber != nil) {
+      url = [NSString stringWithFormat:@"%@&prefill_phone_number=%@", url, registrationSettings.prefillPhoneNumber];
+    }
+  }
```
Comment (lilach) [inline KlarnaOnDemand/KODUrl.m]: create a mutable string and append the data to it

File: KlarnaOnDemandTests/KODUrlSpec.m [hunk 4]
```diff
@@ -4,56 +4,69 @@
...
   });
+
+  it(@"should include confirmed user data id in the registration url when supplied", ^{
+    KODRegistrationSettings *registrationSettings = [[KODRegistrationSettings alloc] initWithPrefillPhoneNumber:nil andConfirmedUserDataId:@"abcd"];
+    NSString *registrationUrl = [KODUrl registrationUrlWithSettings:registrationSettings].absoluteString;
+    [[registrationUrl should] containString: [NSString stringWithFormat:@"confirmed_user_data_id=abcd"]];
+  });
+
+  it(@"should include phone number in the registration url when supplied", ^{
+      KODRegistrationSettings *registrationSettings = [[KODRegistrationSettings alloc] initWithPrefillPhoneNumber:@"12345678" andConfirmedUserDataId:nil];
```
Comment (lilach) [inline KlarnaOnDemandTests/KODUrlSpec.m]: space invasion :smile_cat:

File: KlarnaOnDemandTests/KODUrlSpec.m [hunk 5]
```diff
@@ -4,56 +4,69 @@
...
+
+    NSString *registrationUrl = [KODUrl registrationUrlWithSettings:registrationSettings].absoluteString;
+    [[registrationUrl should] containString: [NSString stringWithFormat:@"prefill_phone_number=12345678"]];
+  });
```
Comment (lilach) [inline KlarnaOnDemandTests/KODUrlSpec.m]: missing spec for both prefilPhoneNumber and confirmedUserDataId

**Comment by lilach** [comment]
add example use in sample app

**Comment by lilach** [comment]
don't forget to change the version (and maybe create a changelog..)

File: KlarnaOnDemand/KODRegistrationViewController.h (hunk 2 above)
Comment (tomerlevi444) [inline KlarnaOnDemand/KODRegistrationViewController.h]: done

File: KlarnaOnDemand/KODRegistrationViewController.h (hunk 1 above)
Comment (tomerlevi444) [inline KlarnaOnDemand/KODRegistrationViewController.h]: done

File: KlarnaOnDemand/KODUrl.m (hunk 3 above)
Comment (tomerlevi444) [inline KlarnaOnDemand/KODUrl.m]: done

File: KlarnaOnDemandTests/KODUrlSpec.m (hunk 4 above)
Comment (tomerlevi444) [inline KlarnaOnDemandTests/KODUrlSpec.m]: done

File: KlarnaOnDemandTests/KODUrlSpec.m (hunk 5 above)
Comment (tomerlevi444) [inline KlarnaOnDemandTests/KODUrlSpec.m]: done

**Pull request merged**